dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ClientBase.cs

```diff
@@ -340,12 +344,12 @@ await foreach (Completions completions in response)
             StringBuilder? contentBuilder = null;
             string? functionName = null;
             StringBuilder? functionArgumentsBuilder = null;
-            ChatRole streamedRole = default;
+            ChatRole? streamedRole = default;
```

Comment:
Why is this being made nullable?

Reply:
To make it easier to use the less verbose `??=` operator. If this should be avoided I'm ok in adding an if statement in the foreach.

Reply:
I'd done this matching the README example at https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/openai/Azure.AI.OpenAI#use-chat-functions (the last one with streaming).  Is it wrong, or did I copy it wrong? My understanding was that the SDK ensures that once Role shows up, subsequent updates include that same Role, so it's ok to always use the last. Is that not the case?

Reply:
Seems to be wrong in the their example too.

This breaks when we try to reuse it for the next client.Call
Basically, the role only comes one chunk, 

`streamdRole = update.Role ?? default`
(will update the role to null again, erasing the chunk that has it set)

[This is a working sample of them](https://github.com/Azure/azure-sdk-for-net/blob/bfc10208f4abfb0d44463d64808132c2f68e5915/sdk/openai/Azure.AI.OpenAI/tests/Samples/Sample04_StreamingChat.cs#L33) on how to handle Roles.

[The function calling sample does not do the same](https://github.com/Azure/azure-sdk-for-net/blob/bfc10208f4abfb0d44463d64808132c2f68e5915/sdk/openai/Azure.AI.OpenAI/tests/Samples/Sample07_ChatFunctions.cs#L117C17-L117C55)

Will file an issue there. Thanks for letting me know.

Reply:
Ok, thanks. I really thought the Azure SDK was propagating the role from update to update. Maybe that got dropped when they changed the streaming model? (Or maybe I'm completely misremembering.)